cocos/ui/UIScale9Sprite.cpp

```diff
@@ -84,17 +84,25 @@ namespace ui {
     bool Scale9Sprite::initWithSpriteFrame(SpriteFrame* spriteFrame,
                                            const Rect& capInsets)
     {
-        Texture2D* texture = spriteFrame->getTexture();
-        CCASSERT(texture != NULL, "CCTexture must be not nil");
-        Sprite *sprite = Sprite::createWithSpriteFrame(spriteFrame);
-        CCASSERT(sprite != NULL, "sprite must be not nil");
-        bool pReturn = this->init(sprite,
-                                  spriteFrame->getRect(),
-                                  spriteFrame->isRotated(),
-                                  spriteFrame->getOffset(),
-                                  spriteFrame->getOriginalSize(),
-                                  capInsets);
-        return pReturn;
+        bool ret = false;
```

Comment:
ret should be `true` and before break, we should set `ret = false`.
otherwise, the return value would always be false.